build.rs

```diff
@@ -0,0 +1,56 @@
+use rerun_except::rerun_except;
+
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+static BOEHM_REPO: &str = "git://github.com/ivmai/bdwgc.git";
+static BOEHM_ATOMICS_REPO: &str = "git://github.com/ivmai/libatomic_ops.git";
+static BOEHM_DIR: &str = "bdwgc";
+static POINTER_MASK: &str = "-DPOINTER_MASK=0xFFFFFFFFFFFFFFF8";
+
+fn run<F>(name: &str, mut configure: F)
+where
+    F: FnMut(&mut Command) -> &mut Command,
+{
+    let mut command = Command::new(name);
+    let configured = configure(&mut command);
+    println!("Executing {:?}", configured);
+    if !configured.status().is_ok() {
+        panic!("failed to execute {:?}", configured);
+    }
+    println!("Command {:?} finished successfully", configured);
+}
+
+fn main() {
+    rerun_except(&["gc_tests/tests/*.rs"]).unwrap();
+
+    let mut bdwgc = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    bdwgc.push("target");
+    #[cfg(debug_assertions)]
+    bdwgc.push("debug");
+    #[cfg(not(debug_assertions))]
+    bdwgc.push("release");
+
+    let mut bdwgc_libs = PathBuf::from(&bdwgc);
+    bdwgc_libs.push("lib");
+
+
+    if bdwgc_libs.exists() {
+        return
+    }
+
+    if !Path::new(BOEHM_DIR).exists() {
+        run("git", |cmd| cmd.arg("clone").arg(BOEHM_REPO));
+
+        run("git", |cmd| {
+            cmd.arg("clone").arg(BOEHM_ATOMICS_REPO).current_dir(BOEHM_DIR)
+        });
+    }
+
+    env::set_current_dir(BOEHM_DIR).unwrap();
+
+    run("./autogen.sh", |cmd| cmd);
+    run("./configure", |cmd| cmd.arg(format!("--prefix={}", bdwgc.to_str().unwrap())).env("CFLAGS", POINTER_MASK));
+    run("make", |cmd| cmd.arg("-j").arg("install"));
```

Comment:
`make -j` is comically dangerous and it's insane that `gmake` offers it! [it's also not portable -- BSD make requires an integer for `-j`.] I suggest limiting this to the number of cpus. I use the `num_cpus` crate for this.

Reply:
I've pushed a couple of commits which simplify the build script to use `OUT_DIR`. As part of this they use `num_cpus` with `make -j` which hopefully fixes the BSD portability issues.